File: 백준/Silver/1065. 한수/한수.py
import sys


def input(): return sys.stdin.readline().strip()


N = int(input())
count = 0

for num in range(1, N + 1):
    num_str = str(num)
    if len(num_str) <= 2:
        count += 1
    else:
        is_hansu = True
        diff = int(num_str[1]) - int(num_str[0])
        for i in range(2, len(num_str)):
            if int(num_str[i]) - int(num_str[i - 1]) != diff:
                is_hansu = False
                break
        if is_hansu:
            count += 1

print(count)

File: 백준/Silver/1065. 한수/한수.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

bool is_hansu(int num)
{
    char num_str[5];
    sprintf(num_str, "%d", num);

    if (strlen(num_str) <= 2)
    {
        return true;
    }
    else
    {
        int diff = num_str[1] - num_str[0];
        for (int i = 2; i < strlen(num_str); i++)
        {
            if (num_str[i] - num_str[i - 1] != diff)
            {
                return false;
            }
        }
    }
    return true;
}

int main()
{
    int N;
    scanf("%d", &N);

    int count = 0;
    for (int num = 1; num <= N; num++)
    {
        if (is_hansu(num))
        {
            count++;
        }
    }

    printf("%d\n", count);
    return 0;
}